test/net_http_test.rb

```diff
@@ -312,6 +312,43 @@ def test_adding_custom_errors_do_trip_circuit
     end
   end
 
+  def test_5xxs_trip_circuit_when_fatal_server_flag_enabled
+    options = proc do |host, port|
+      {
+        tickets: 2,
+        success_threshold: 1,
+        error_threshold: 3,
+        error_timeout: 10,
+        open_circuit_server_errors: true,
+        name: "#{host}_#{port}",
+      }
+    end
+
+    with_semian_configuration(options) do
+      with_server do
+        http = Net::HTTP.new(HOSTNAME, PORT)
+        http.raw_semian_options[:error_threshold].times do
+          http.get("/500")
+        end
+        assert_raises Net::CircuitOpenError do
```

Comment:
That's the model I went for in the MySQL adapter with `#ping`, where it'll return `false` if the circuit is open as the fallback—not raise an exception. I don't have a strong opinion on this.